frontend/apps/crates/utils/src/ages.rs

```diff
@@ -7,19 +7,29 @@ pub trait AgeRangeVecExt {
 
 impl AgeRangeVecExt for Vec<AgeRange> {
     fn range_string(&self, selected: &Vec<AgeRangeId>) -> String {
-        let first_age = self.iter().find(|age| selected.contains(&age.id));
+        if selected.len() == self.len() {
+            "All Ages".to_string()
+        } else {
+            let first_age = self
+                .iter()
+                .find(|age| selected.contains(&age.id))
+                .unwrap();
+            let last_age = self
+                .iter()
+                .rev()
+                .find(|age| selected.contains(&age.id))
+                .unwrap();
 
-        let mut result = String::new();
-        if let Some(first_age) = first_age {
-            result = first_age.display_name.clone();
-        };
+            let first_age_text = first_age.short_display_name.as_ref().unwrap_or(&first_age.display_name);
```

Comment:
`short_display_name` is an `Option`. Using `display_name` as fallback if `short_display_name` is None.